server/auth/registrar.go

```diff
@@ -25,7 +27,320 @@ var (
+	if err != nil {
+		return msgjson.NewError(msgjson.BondError, "invalid bond transaction: %v", err)
+	}
+	if amt < int64(auth.minBond) {
+		return msgjson.NewError(msgjson.BondError, "insufficient bond amount %d, needed %d", amt, auth.minBond)
+	}
+	if lockTime < lockTimeThresh.Unix() {
+		return msgjson.NewError(msgjson.BondError, "insufficient lock time %d, needed at least %d", lockTime, lockTimeThresh.Unix())
+	}
+
+	// Must be equal to account ID computed from pubkey in the PayFee message.
+	if commitAcct != acctID {
+		return msgjson.NewError(msgjson.BondError, "invalid bond transaction - account commitment does not match pubkey")
+	}
+
+	// All good. The client gets a PostBondResult (no error) unless the confirms
+	// check has an unexpected error or times out.
+	expireTime := time.Unix(lockTime, 0).Add(-auth.bondExpiry)
+	postBondRes := &msgjson.PostBondResult{
+		AccountID: acctID[:],
+		AssetID:   assetID,
+		Amt:       uint64(amt),
+		Expiry:    expireTime.Unix(),
+		BondID:    bondCoinID,
+		// Confs is set below.
+	}
+	auth.Sign(postBondRes)
+
+	sendResp := func() *msgjson.Error {
+		resp, err := msgjson.NewResponse(msg.ID, postBondRes, nil)
+		if err != nil { // shouldn't be possible
+			return msgjson.NewError(msgjson.RPCInternalError, "internal encoding error")
+		}
+		err = conn.Send(resp)
+		if err != nil {
+			log.Warnf("Error sending postbond result to user %v: %v", acctID, err)
+			if err = auth.Send(acctID, resp); err != nil {
+				log.Warnf("Error sending postbond result to account %v: %v", acctID, err)
+				// The user will need to 'connect' to reconcile bond status.
+			}
+		}
+		return nil
+	}
+
+	// See if the account exists, and get known unexpired bonds. Also see if the
+	// account has previously paid a legacy registration fee.
+	dbAcct, bonds, _, _ := auth.storage.Account(acctID, lockTimeThresh)
+	newAcct := dbAcct == nil
+
+	reqConfs := int64(auth.bondConfs)
+
+	// See if we already have this bond in DB.
+	for _, bond := range bonds {
+		if bond.AssetID == assetID && bytes.Equal(bond.CoinID, bondCoinID) {
+			postBondRes.Active = true
+			_, postBondRes.Tier = auth.AcctStatus(acctID)
+			return sendResp()
+		}
+	}
+
+	bondStr := coinIDString(assetID, bondCoinID)
+	bondAssetSym := dex.BipIDSymbol(assetID)
+
+	// Determine if it exists and is confirmed.
+	chkAddr, chkAmt, confs, err := auth.checkFee(assetID, bondCoinID)
+	if err != nil {
+		// Not found is expected if this is a pre-broadcast validation request.
+		if !errors.Is(err, asset.CoinNotFoundError) {
+			return msgjson.NewError(msgjson.BondError, "unable to check bond with node: %v", err)
+		}
+
+		// OK. Respond with active=false. They may now broadcast, wait for the
+		// required confirms, and postbond again.
+		log.Debugf("Validated prospective bond txn output %s (%s) paying %d for user %v",
+			bondStr, bondAssetSym, amt, acctID)
+		postBondRes.Active = false
+		return sendResp()
+	}
```

Comment:
This makes the latency queue somewhat useless. The latency queue is convenient because the user can submit their transaction and the communicate with the server immediately, expecting that if the server does not see the tx yet, they will wait for it. Now, if the server doesn't see the tx yet, they will assume this is a pre-validation and won't start the queue.

Reply:
Oh. Nevermind. Client must wait for confs before sending. Doof.